### PR TITLE
Add dataexplorer tests

### PR DESCRIPTION
**Pull request by ambrussimon-invenshure**
Added dataexplorerhandler unit test with mocked ElasticSearch.
@nagem @hkethi002:
* gathered the source fields into module level vars
* found unreachable lines (added `#NOTE unreachable`)
* didn't get to _handle_properties recursion yet
* didn't get to facet_status @ indexing yet

### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

File: api/handlers/dataexplorerhandler.py [hunk 1]
```diff
@@ -308,6 +354,7 @@ def aggregate_field_values(self):
             }
         }
         if not filters:
+            # NOTE unreachable
```
Comment (nagem) [inline api/handlers/dataexplorerhandler.py]: The code will need to get updated to support device behavior for completeness. That's where an empty filter could come from (no `uid` for device auth), but it doesn't currently support it. I'll open a ticket about supporting device requests in search.

File: api/handlers/dataexplorerhandler.py [hunk 2]
```diff
@@ -374,6 +421,7 @@ def search_fields(self):
         try:
             field_query = str(field_query)
         except ValueError:
+            # NOTE unreachable
             self.abort(400, 'Must specify string for field query')
```
Comment (nagem) [inline api/handlers/dataexplorerhandler.py]: This can get removed

File: api/handlers/dataexplorerhandler.py [hunk 3]
```diff
@@ -228,6 +228,52 @@
 }
 
 
+SOURCE_COMMON = [
```
Comment (nagem) [inline api/handlers/dataexplorerhandler.py]: Thanks for cleaning this up 👍

File: api/handlers/dataexplorerhandler.py (hunk 2 above)
Comment (ambrussimon-invenshure) [inline api/handlers/dataexplorerhandler.py]: Do you mean the whole except or just the comment? :)

**Comment by nagem** [comment]
@ambrussimon-invenshure feel free to merge this one.

File: api/handlers/dataexplorerhandler.py (hunk 2 above)
Comment (nagem) [inline api/handlers/dataexplorerhandler.py]: I meant the try/except around the `str` cast.

**Pull request merged**